ext/AvizExt/viz/location_selection.jl

```diff
@@ -102,32 +104,134 @@ function ADRIA.viz.ranks_to_frequencies!(
-        AxisKeys.keyless(NamedDims.unname(frequencies[ranks=rank_id]));
-        opts=opts,
-        axis_opts=axis_opts,
+        AxisKeys.keyless(NamedDims.unname(frequencies[ranks = rank_id]));
+        opts = opts,
+        axis_opts = axis_opts,
     )
 end
 function ADRIA.viz.ranks_to_frequencies(
     rs::ResultSet,
     frequencies::NamedDimsArray,
-    rank_ids::Union{Int64,Vector{Int64}};
-    opts::Dict=Dict(),
-    fig_opts::Dict=Dict(), axis_opts::Dict=Dict())
-
+    rank_ids::Union{Int64, Vector{Int64}};
+    opts::Dict = Dict(),
+    fig_opts::Dict = Dict(), axis_opts::Dict = Dict())
     f = Figure(; fig_opts...)
     g = f[1, 1] = GridLayout()
     ADRIA.viz.ranks_to_frequencies!(
         g,
         rs,
         frequencies,
         rank_ids;
-        opts=opts,
-        axis_opts=axis_opts,
+        opts = opts,
+        axis_opts = axis_opts
     )
 
     return f
 end
 
+"""
+    ADRIA.viz.decision_matrices(rs::ResultSet, S::NamedDimsArray, scores::Vector{Float64}, 
+        criteria::Vector{Symbol}; opts::Dict=Dict(), axis_opts::Dict=Dict(), fig_opts::Dict=Dict())
+    ADRIA.viz.decision_matrices!(g::Union{GridLayout,GridPosition}, rs::ResultSet, 
+        S::NamedDimsArray, scores::Vector{Float64}, criteria::Vector{Symbol}; 
+        opts::Dict=Dict(), axis_opts::Dict=Dict())
+        
+Plot a grid of spatial maps for a selection of location selection criteria.
+
+# Arguments
+- `g` : Figure GridPosition or GridLayout
+- `rs` : Result set
+- `S` : A normalised decision matrix calculated using decison.decision_matrices
+- `criteria` : Names of criteria to be plotted, e.g [:heat_stress, :wave_stress]
+- `opts` : Aviz options
+    - `colorbar_label`, label for colorbar. Defaults to "Relative Cover".
+    - `color_map`, preferred colormap for plotting heatmaps.
+    - `colorbar_limits` : Upper and lower limits displayed on colorbar,
+    (default is (0.0, 1.0))
+- `axis_opts` : Additional options to pass to adjust Axis attributes
+  See: https://docs.makie.org/v0.19/api/index.html#Axis
+- `fig_opts` : Additional options to pass to adjust Figure creation
+  See: https://docs.makie.org/v0.19/api/index.html#Figure
+
+# Returns
+Figure
+"""
+function ADRIA.viz.decision_matrices(
+    rs::ResultSet,
+    S::NamedDimsArray,
+    scores::Vector{Float64},
+    criteria::Vector{Symbol};
+    opts::Dict = Dict(),
+    axis_opts::Dict = Dict(),
+    fig_opts::Dict = Dict(),
+)
+    f = Figure(; fig_opts...)
+    g = f[1, 1] = GridLayout()
+    ADRIA.viz.decision_matrices!(
+        g, rs, S, scores, criteria; opts = opts, axis_opts = axis_opts
+    )
+    return f
+end
+function ADRIA.viz.decision_matrices!(
+    g::Union{GridLayout, GridPosition},
+    rs::ResultSet,
+    S::NamedDimsArray,
+    scores::Vector{Float64},
+    criteria::Vector{Symbol};
+    opts::Dict = Dict(),
+    axis_opts::Dict = Dict(),
+)
+    if length(rs.site_data.site_id) != size(S, 1)
+        error("Only unfiltered decision matrices can be plotted.")
```

Comment:
The decision matrices can be filtered based on thresholds for different criteria (like heat stress and wave stress) but this means they have less sites than the whole site set, which means when you go to plot them the geodata (created from the result set) doesn't match the number of sites when plotting using `map`. Alternatively, filtered sites could be set as `0.0` or something, but they'd have to be added back into the decision matrix because these are automatically filtered in `create_decision_matrix`.